enterprise/backend/src/metabase_enterprise/serialization/v2/extract.clj

```diff
@@ -60,16 +79,21 @@
         cards          (reduce set/union (for [coll-id collection-set]
                                            (db/select-ids Card :collection_id coll-id)))
 
-        ;; Map of {dashboard-id #{DashboardCard}} for dashcards whose cards are outside the transitive collection set.
+        ;; Map of {dashboard-id #{DashboardCard}} for dashcards whose cards OR parameter-bound cards are outside the
+        ;; transitive collection set.
         escaped-dashcards  (into {}
                                  (for [dash  dashboards
                                        :let [dcs (db/select DashboardCard :dashboard_id (:id dash))
                                              escapees (->> dcs
-                                                           (filter :card_id) ; Text cards have a nil card_id
-                                                           (filter (comp not cards :card_id))
-                                                           set)]
-                                       :when (seq escapees)]
-                                   [(:id dash) escapees]))
+                                                           (keep :card_id) ; Text cards have a nil card_id
+                                                           set)
+                                             params   (->> dcs
+                                                           (mapcat :parameter_mappings)
+                                                           (keep :card_id)
+                                                           set)
+                                             combined (set/difference (set/union escapees params) cards)]
+                                       :when (seq combined)]
+                                   [(:id dash) combined]))
```

Comment:
Could you fetch all the DashboardCards for all the relevant Dashboards in a single QB query? 

```clj
(db/select DashboardCard :dashboard_id [:in (set (map :id dashboards))])
```

and then `group-by` `:dashboard_id` or something. If you have 40 dashboards you don't want to have to do 40 extra queries to get DashboardCards